src/main/scala/io/github/exploiting/flinks/stateful/operators/advanced_fc/AdvancedStatefulFlyingCarpetJob.scala

```diff
@@ -119,42 +126,37 @@ class JoinStreams extends RichCoFlatMapFunction[UuidTransaction, ResultEvent, En
   override def flatMap1(value: UuidTransaction, out: Collector[EnrichedTransaction]): Unit = {
     val expectedResults = value.splitUp
 
-    val results = Option(resultEvents.value()).getOrElse(Map.empty)
-
-    if (results.size == expectedResults) {
+    if (numberOfResultEvents == expectedResults) {
       // We're complete, collect.
-      out.collect(EnrichedTransaction(value.id, value.payload, results))
+      out.collect(EnrichedTransaction(value.id, value.payload, resultEventsAsScalaMap))
       resultEvents.clear()
       enrichedEvent.clear()
     }
     else {
-      enrichedEvent.update(Seq(EnrichedTransaction(value.id, value.payload, results)))
+      enrichedEvent.add(EnrichedTransaction(value.id, value.payload, resultEventsAsScalaMap))
```

Comment:
i think there is no point of storing the results in the EnrichedTransaction at this point, the results are stored in the results map, and we can add them to EnrichedTransaction just before  collecting it.
Also, i think the enrichedEvent state should just be of type UuidTransaction